### PR TITLE
Do not always add NewLine following MultiLineComments.

### PR DESCRIPTION
**Pull request by JoeRobich**
Resolves https://github.com/dotnet/roslyn/issues/39351

Allow the adjust newlines operation to dictate how many lines to add following a multiline comment.

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs [hunk 1]
```diff
@@ -72,14 +72,15 @@ protected override LineColumnRule GetLineColumnRuleBetween(SyntaxTrivia trivia1,
             // [trivia] [whitespace] [token] case
             if (trivia2.IsKind(SyntaxKind.None))
             {
-                var insertNewLine = this.FormattingRules.GetAdjustNewLinesOperation(this.Token1, this.Token2) != null;
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs:75]: ❔ Do you know why this is coming back as non-null?

Comment (JoeRobich) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs:75]: Seems we always generate it when we hit the EndOfLine token. Should that controlled by the `insert_final_newline` .editorconfig option?

https://github.com/dotnet/roslyn/blob/65d799af3a796c7928c5a6acfc961a15c9916d5f/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/EndOfFileTokenFormattingRule.cs#L15-L25

Comment (sharwell) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs:75]: I don't think we need to make a change to that code _yet_. Do we know why `IsStartOrEndOfFile` (a few lines above) isn't returning true for this case?

Comment (JoeRobich) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs:75]: https://github.com/dotnet/roslyn/blob/5b1cdefccd312a1b88e7e85a62d9ba918b87815e/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs#L167-L168

Kinds:
trivia1 - MultiLineCommentTrivia (8542)
trivia2 - None (0)
Token1 - SemicolonToken (8212)
Token2 - EndOfFileToken (8496)

Comment (JoeRobich) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs]: @sharwell Updated IsStartOrEndOfFile to check for EndOfFileToken and reverted earlier changes.

**Pull request merged**